src/askvm/resources/core/assign.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  any,
+  resource,
+  run,
+  runUntyped,
+  Options,
+  TypedValue,
+  JSONable,
+} from '../../lib';
+
+function assignValue(
+  valueObject: { [key: string]: any },
+  key: string,
+  assignedValue: any
+) {
+  if (Object.isFrozen(valueObject[key]))
+    throw new Error(`Cannot assign to a constant variable "${key}"`);
+  valueObject[key] = assignedValue;
+}
+
+export const assignRes = resource({
+  type: any,
+  async compute(options, code, args): Promise<TypedValue<JSONable>> {
+    const { params: children = [] } = code;
+
+    const key: any = await runUntyped(options, children[0]); // FIXME any
+    const value = await run(options, children[1]);
+
+    if (key === 'resources') {
+      throw new Error(`Key "resources" cannot be redeclared`);
+    }
+
+    let keyFound = false;
+    // if (code.name === 'assign') {
+    for (
+      let prototype: Options | undefined = options;
+      prototype;
+      prototype = prototype?.prototype
+    ) {
+      const { values = {} } = prototype;
+      if (Object.prototype.hasOwnProperty.call(values, key)) {
+        assignValue(values, key, value);
+        keyFound = true;
+        return value;
+      }
+    }
+    if (keyFound === false)
```

Comment:
This condition is not tested - please a test for it

Reply:
Implemented